spirv_cross.cpp

```diff
@@ -3910,10 +3928,40 @@ void Compiler::CombinedImageSamplerUsageHandler::add_hierarchy_to_comparison_ids
 {
 	// Traverse the variable dependency hierarchy and tag everything in its path with comparison ids.
 	comparison_ids.insert(id);
+	
+	// If the underlying resource has been used for comparison then duplicate loads of that resource must be too.
+	if (!compiler.supports_combined_samplers())
+	{
+		for (auto it = dependency_hierarchy.begin(); it != dependency_hierarchy.end(); ++it)
```

Comment:
Since most of the changes are not from me, I still need to grasp a few things myself. Maybe you can clarify a few things for me, for example what is the type's meaning of `dependency_hierarchy`? It was already part of SPIRV-Cross, but the code doesn't document what this map-to-map type is supposed to accomplish.

Regarding your question:
As far as I remember, this fixed an issue from a Vulkan GLSL to ESSL translation when a texture was used with a regular sampler and a comparison sampler which needs to be merged into combined-texture-samplers for ESSL. But I'm not 100% sure.

Reply:
The dependency hierarchy is used in combined image sampler analysis to figure out which samplers and textures need to be marked as such. We need to be able to trace back from a sampling instruction or OpSampledImage all the way back to the original image, even through call stacks. I.e. a call-site -> image traversal.

From what I can understand of this, it seems to propagate this information the other way, from image to call-site. From image back up to the combined sampler call-site?